src/reducers/global.ts

```diff
@@ -66,11 +67,9 @@ const defaultState = {
     fetchInProcess: false, // App is fetching data (takes a few seconds)
     targetUrl: "", // URL we were given to fetch
     errorDuringFetch: false, // Was there an error around fetch process
-    errorMessage: "", // If error, human-readable message about the error.
-    displayError: false, // Should we display error about fetch to user?
   },
   treeTitle: "",
-  showTreeFormatError: false,
+  showErrorMessages: showErrorDefaults,
```

Comment:
I'm not advocating for it, but this is where you might consider breaking out a new redux reducer "bucket". We discussed awhile ago how it's usually convenient to have all your redux stuff live in a single reducer so you can work on everything in one place. But if you start to see yourself breaking things out because they never talk to each other and the abstraction is much easier to think about (like how you have `showErrorDefaults` being imported from its own file rather than living here), that's when to consider breaking it out. It's likely error stuff would never communicate back to the main set of info, so it would be a sane choice.

I'm really not saying you should do it now, but something to think about as it grows. If there's a really clear dividing line and you never think you'll have to reach across that line ever, it can make it easier to follow the logic.

Reply:
Yeah, the mega-reducer is definitely starting to feel a bit unwieldy -- I think it's OK to leave as-is for now, but agree that if we decide to continue adding features, etc, it's a good eng excellence project to refactor this in the medium-term future.